src/auth/user/user.service.ts

```diff
@@ -56,6 +65,25 @@ export class UserService {
       return user;
     } catch (error) {
       Logger.error('Error verifying user', error);
+      throw new HttpException(
+        error.message || ERROR_MESSAGES.INTERNAL_SERVER_ERROR,
+        error.status || HttpStatus.INTERNAL_SERVER_ERROR,
+      );
+    }
+  }
+
+  async isUserExistsCheck(username: string, email: string): Promise<boolean> {
+    try {
+      const user = await this.userRepository.findOne({
```

Comment:
Change function name to validateCreateUserDto since we are attempting to validate it here.